src/github.com/matrix-org/dendrite/roomserver/input/events.go

```diff
@@ -79,8 +76,8 @@ func processRoomEvent(db RoomEventDatabase, ow OutputRoomEventWriter, input api.
 		if input.HasState {
 			// We've been told what the state at the event is so we don't need to calculate it.
 			// Check that those state events are in the database and store the state.
-			entries, err := db.StateEntriesForEventIDs(input.StateEventIDs)
-			if err != nil {
+			var entries []types.StateEntry
+			if entries, err = db.StateEntriesForEventIDs(input.StateEventIDs); err != nil {
```

Comment:
Humm. I'm not sure I like this. Semantically this diff is the same but style-wise I dislike it. I dislike it because, at a glance, I thought `entries` was scoped to just the `if` block due to the inlining. It's not, since there is a `=` and not `:=`, but looking for a single `:` on this long if statement is harder than just having it what it was before.

I don't mind us inlining `expression; conditional` provided the expression just returns an `err` since that scopes things nicely. WDYT?